runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryPanel.java

```diff
@@ -144,16 +158,84 @@ void update()
 
 			for (Map.Entry<Book, BookPanel> e : bookPanels.entrySet())
 			{
-				HashSet<String> locs = bookLocations.get(e.getKey());
-				if (locs == null || locs.size() > 3)
+				if (e.getKey().isDarkManuscript())
 				{
-					e.getValue().setLocation("Unknown");
+					if (manuscriptLocation == null)
+					{
+						e.getValue().setLocation("Unknown");
+						library.setManuscriptDirections(false);
+					}
+					else
+					{
+						e.getValue().setLocation("<html>" + getFormattedLocation(manuscriptLocation) + "</html>");
+					}
```

Comment:
Do you mean in the panel? Because that would be all rooms/locations. My opinion is that it is better to list the location as "Unknown" in the panel. If you mean the tile markers, they will all show unless the config is checked to hide unconfirmed locations.

Reply:
It's possible to have certain areas without any dark manuscripts in the current rotation, so it's not all rooms. It might be better to put them down at the bottom though, to make it less spammy.